Common/Utils/include/CommonUtils/RootSerializableKeyValueStore.h

```diff
@@ -0,0 +1,175 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef ALICEO2_ROOTSERKEYVALUESTORE_H
+#define ALICEO2_ROOTSERKEYVALUESTORE_H
+
+#include <map>
+#include <string>
+#include <TClass.h>
+#include <Rtypes.h>
+#include <typeinfo>
+#include <typeindex>
+#include <iostream>
+#include <TBufferFile.h>
+#include <type_traits>
+
+namespace o2
+{
+namespace utils
+{
+
+/// A ROOT serializable container mapping a key (string) to an object of arbitrary type.
+///
+/// The container allows to group objects in heterogeneous type in a key-value container.
+/// The container can be ROOT serialized which adds on-top of existing solutions such as boost::property_tree.
+/// This may be useful in various circumenstances, for
+/// instance to assemble various CCDB objects into a single aggregate to reduce the number of CCDB files/entries.
+class RootSerializableKeyValueStore
+{
+ public:
+  /// Structure encapsulating the stored information: raw buffers and attached type information (combination of type_index_hash and TClass information)
+  struct SerializedInfo {
+    void* objptr = nullptr; //! pointer to existing object in memory
+    Int_t N = 0;
+    char* bufferptr = nullptr; //[N]  pointer to serialized buffer
+
+    // we use the TClass and/or the type_index_hash for type idendification
+    TClass* cl = nullptr;
+    std::string typeinfo_name; // typeinfo name that can be used to store type if TClass not available (for PODs!)
+    ClassDefNV(SerializedInfo, 1);
+  };
+
+  RootSerializableKeyValueStore() = default;
+
+  /// putting a value
+  /// T needs to be trivial non-pointer type or a type having a ROOT TClass instance
+  template <typename T>
+  void put(std::string const& key, T const& value)
+  {
+    return put_impl(key, value, int{});
+  }
+
+  /// returns object for this key or nullptr if error or does not exist
+  /// TODO: error handling/info via some enum?
+  template <typename T>
+  const T* get(std::string const& key) const
```

Comment:
ciao @sawenzel 
looks like the machinery works.
I wonder though why the `get` call returns a pointer.
Could it return by reference?
```
const T& get(std::string const& key) const
```

Reply:
in case it is to be able to tell if the key exists, perhaps one can just add a `bool has(std::string const& key) const` function to check for the existence of it?

Reply:
uhm... maybe I see the problem.
what reference to return when the key does not exist?

Reply:
nothing that cannot be resolved, in principle
```
  template <typename T>
  const T& get2(std::string const& key) const
  {
    auto ptr = get_impl<T>(key, int{});
    if (!ptr) {
      static T val = T();
      return val;
    }
    return *ptr;
  }
```
this works also for not-existing keys.

Reply:
the function can be overloaded to provide hints on the status of the returned object (good, problems)
```
  template <typename T>
  const T& get2(std::string const& key, int& errflag) const
  {
    auto ptr = get_impl<T>(key, int{});
    if (!ptr) {
      static T val = T();
      errflag = 1;
      return val;
    }
    errflag = 0;
    return *ptr;
  }
```